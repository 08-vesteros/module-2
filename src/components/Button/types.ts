import { MouseEventHandler } from 'react';

export type ButtonProps = {
	onClick: MouseEventHandler<HTMLButtonElement>;
	content: string;
};

export type ButtonWrapperProps = {
	disabled?: boolean;
};

export type Props = ButtonProps & ButtonWrapperProps;
