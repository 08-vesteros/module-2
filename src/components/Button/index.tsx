import React, { FC } from "react";


const Button: FC = ({children}) => (
            <ButtonWrapper>{children}</ButtonWrapper>
      );
  
  export default Button;