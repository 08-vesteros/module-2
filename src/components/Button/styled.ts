import styled from 'styled-components';
import {
	PRIMARY_COLOR,
	DARK_PRIMARY_COLOR,
	BLACK_COLOR,
	WHITE_COLOR,
} from '../../constants';
import { ButtonWrapperProps } from './types';

export const inGameButtonTheme = {
	backgroundColor: PRIMARY_COLOR,
	border: `2px solid ${PRIMARY_COLOR}`,
	backgroundColorHover: DARK_PRIMARY_COLOR,
	color: WHITE_COLOR,
};

export const ButtonWrapper = styled.button<ButtonWrapperProps>`
	cursor: pointer;
	border-radius: 8px;
	background-color: ${props => props.theme.backgroundColor || 'transparent'};
	border: ${props => props.theme.border || `2px solid ${PRIMARY_COLOR}`};
	color: ${props => props.theme.color || BLACK_COLOR};
	padding: ${props => props.theme.padding || '12px 0'};
	margin: ${props => props.theme.margin || '0 0 24px 0'};
	width: ${props => props.theme.width || '180px'};
	opacity: ${props => (props.disabled ? 0.3 : 1)};
	pointer-events: ${props => (props.disabled ? 'none' : 'auto')};
	font-size: 16px;

	&:hover {
		background-color: ${props =>
			props.theme.backgroundColorHover || PRIMARY_COLOR};
		color: ${props => props.theme.colorHover || WHITE_COLOR};
		transition: all 0.3s;
	}
`;
