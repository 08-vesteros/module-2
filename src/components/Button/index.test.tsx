import * as React from 'react';
import { shallow } from 'enzyme';
import Button from '.';

describe('Button component', () => {
	it('should render component', () => {
		const component = shallow(<Button content='start' />);
		expect(component).toMatchSnapshot();
	});

	it('should call onClick method', () => {
		const mockCallBack = jest.fn();
		const component = shallow(<Button content='start' onClick={mockCallBack} />);
		expect(mockCallBack.mock.calls.length).toBe(0);
		component.simulate('click');
		expect(mockCallBack.mock.calls.length).toBe(1);
	});
});
