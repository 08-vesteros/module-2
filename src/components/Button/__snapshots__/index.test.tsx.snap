// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render component 1`] = `
<styled.button>
  start
</styled.button>
`;
